core/libraries/rest_api/controllers/model/Read.php

```diff
@@ -43,25 +48,29 @@ class Read extends Base
 
     /**
      * Read constructor.
+     * @param CalculatedModelFields $fields_calculator
      */
-    public function __construct()
+    public function __construct(CalculatedModelFields $fields_calculator)
     {
         parent::__construct();
-        $this->fields_calculator = new CalculatedModelFields();
+        $this->fields_calculator = $fields_calculator;
     }
 
 
     /**
      * Handles requests to get all (or a filtered subset) of entities for a particular model
      *
      * @param WP_REST_Request $request
-     * @param string          $version
-     * @param string          $model_name
-     * @return \WP_REST_Response|WP_Error
+     * @param string $version
+     * @param string $model_name
+     * @return WP_REST_Response|WP_Error
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
      */
     public static function handleRequestGetAll(WP_REST_Request $request, $version, $model_name)
     {
-        $controller = new Read();
+        $controller = LoaderFactory::getLoader()->getShared('EventEspresso\core\libraries\rest_api\controllers\model\Read');
```

Comment:
This changes things in that everytime `handleRequestGetAll` is called it is now returning a shared instance of `Read`.  Is that intentional?  Are there any cached things on this class that could cause problems in a long running php worker environment that might use the api internally (such as on ES with gearman)?

**Note: this comment applies everywhere similar behaviour has changed in your work**

Reply:
Yeah it was intentional, but probably an error. There's some debug and response header info that's put onto the read controller, so we should probably be creating a new one instead. So thanks for checking